arrays/arrays.c

```diff
@@ -126,12 +127,12 @@ void arr_append(Array *arr, char *element)
 
   // Resize the array if the number of elements is over capacity
   // or throw an error if resize isn't implemented yet.
-  if (arr->elements == arr->capacity)
+  if (*arr->elements == arr->capacity)
```

Comment:
Shouldn't need an asterisk here unless arr is Array ** (i.e. a double pointer). There could be something specific about your code that needs this though. I'll have to take a closer look.